Source/EasyNetQ/SelfHostedBus.cs

```diff
@@ -1,36 +1,38 @@
-using EasyNetQ.LightInject;
+using System;
+using Microsoft.Extensions.DependencyInjection;
 
-namespace EasyNetQ;
-
-/// <summary>
-/// Bus (in fact wrapper around IBus) with an internal DI container so the caller should dispose it.
-/// </summary>
-public sealed class SelfHostedBus : IBus, IDisposable
+namespace EasyNetQ
 {
-    private readonly IBus bus;
-    private readonly ServiceContainer container;
-
-    internal SelfHostedBus(ServiceContainer container)
+    /// <summary>
+    /// Bus (in fact wrapper around IBus) with an internal DI container so the caller should dispose it.
+    /// </summary>
+    public sealed class SelfHostedBus : IBus, IDisposable
     {
-        this.bus = container.GetInstance<IBus>();
-        this.container = container;
-    }
+        private readonly IBus bus;
+        private readonly IServiceProvider serviceProvider;
 
-    /// <inheritdoc />
-    public void Dispose() => container.Dispose();
+        internal SelfHostedBus(IServiceProvider serviceProvider)
+        {
+            this.serviceProvider = serviceProvider;
+            this.bus = serviceProvider.GetRequiredService<IBus>();
+        }
 
-    /// <inheritdoc />
-    public IPubSub PubSub => bus.PubSub;
+        /// <inheritdoc />
+        public void Dispose() => (serviceProvider as IDisposable)?.Dispose();
```

Comment:
See no issues to inject ServiceProvider (instead of IServiceProvider) directly here to avoid this magic.

Reply:
https://github.com/EasyNetQ/EasyNetQ/pull/1798/commits/da8fb42c21593edb7c565d29f3eee0a92f1f0e1e